pymc3/backends/arviz.py

```diff
@@ -0,0 +1,606 @@
+"""PyMC3-ArviZ conversion code."""
+import logging
+import warnings
+
+from typing import (  # pylint: disable=unused-import
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Tuple,
+    Union,
+)
+
+import numpy as np
+import xarray as xr
+
+from aesara.gof.graph import ancestors
+from aesara.tensor.var import TensorVariable
```

Comment:
It used to check for `PyMC3Variable`, I had some doubts between using `Variable` or `TensorVariable`

Reply:
Usually, the difference comes down to whether or not the code requires shape information (e.g. the `dtype` and `broadcastable` fields that are exposed by `TensorVariable` via `TensorVariable.type`).

Reply:
I don't think we explicitly use this info, but it should be taken into account when converting to numpy to make sure we preserve the dtypes of the original parameters and data. Thanks for the explanation, I will test to make sure dtypes are preserved.